jablib/src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -149,7 +149,8 @@ public Optional<BibEntry> performSearchById(String identifier) throws FetcherExc
             bibtexString = URLDownload.asString(openConnection).trim();
 
             // BibTeX entry
-            fetchedEntry = BibtexParser.singleFromString(bibtexString, preferences);
+            List<BibEntry> entries = new BibtexParser(preferences).parseEntries(bibtexString);
+            fetchedEntry = entries.isEmpty() ? Optional.empty() : Optional.of(entries.getFirst());
```

Comment:
There is no URLconnection passed - please use `org.jabref.logic.importer.fileformat.BibtexParser#parseEntries`

Reply:
I made a typo and meant "stream". Could be seen at the issue description, but maybe for a newcomer, this is too hard to understand.